sipyco/pc_rpc.py

```diff
@@ -507,15 +512,26 @@ def _document_function(function):
             function (Callable): a Python function to be documented.
 
         Returns:
-            Tuple[dict, str]: tuple of (argument specifications,
+            Tuple[str, str]: tuple of (directly rendered argument specifications,
             function documentation).
             Any type annotations are converted to strings (for PYON serialization).
         """
-        argspec_dict = dict(inspect.getfullargspec(function)._asdict())
-        # Fix issue #1186: PYON can't serialize type annotations.
-        if any(argspec_dict.get("annotations", {})):
-            argspec_dict["annotations"] = str(argspec_dict["annotations"])
-        return argspec_dict, inspect.getdoc(function)
+        sig = inspect.signature(function)
+
+        def is_annotation_void(x: typing.Union[inspect.Signature, inspect.Parameter]):
```

Comment:
Type annotations are not used anywhere else in ARTIQ or Sipyco, please remove them.
I'm not generally opposed to type annotations since they are becoming popular with Python and also will be used to great effect in the new ARTIQ compiler, but having a patchwork of annotated/unannotated code sounds messy.
If you really want annotations, please at least annotate the whole module and figure out a way to check the annotations' correctness. This should go in a separate PR in any case.